### DISCUSSION TITLE
T-Deck plus - perfect working touch screen with LGFX/LVGL

### DISCUSSION THREAD

**Discussion by numnim2209** [meshtastic/firmware#5432] [General]
Insert this piece of code:
`// set interrupt mode
        uint8_t buf1[] = { 0x80, 0x4d, 0xb5 };
        _writeBytes(buf1, 3);
        _freshConfig();`

In the file:
`/firmware/.pio/libdeps/t-deck/LovyanGFX/src/lgfx/v1/touch`

at line 83:
![grafik](https://github.com/user-attachments/assets/84de92f7-b700-45ef-b9b1-bef41310e42a)

Before build, change:
variants/t-deck/platformio.ini

![grafik](https://github.com/user-attachments/assets/e7e32879-ef56-4288-a225-16c2152a43a4)

**Comment by numnim2209** [comment]
The filename is:
Touch_GT911.cpp

**Comment by pagex** [comment]
newbie question, where is branch or folder for T Deck Plus to build, I was looking at tft-gui-work branch and environment: t-deck-tft upon building it says  Error: src/main.cpp:118:10: fatal error: DeviceScreen.h: No such file or directory

**Reply by mariotti** [reply]
@pagex I suggest you to follow the standard documentation: https://meshtastic.org/docs/development/firmware/build/ in case you still have issues please start a "new" discussion.

**Comment by mariotti** [comment]
@numnim2209 what started your suggestion?

I did not try it yet but I have issues with the touch screen. Like laggy and not responding to long touch.
Are these the same?

And what this change will do?

As also @pagex is confused, on what branch / PR would this apply?

Thanks for the current suggestion, and in advance for your feed back!

A+
Fabio

**Reply by mverch67** [reply]
When did you purchase that T-Deck Plus? I think only the first batch was affected that shipped in July/August. The newer ones shipped end of September had the fix described on LilyGo's issue page.

**Comment by mariotti** [comment]
Hoi @numnim2209 I created a discussion here: https://github.com/meshtastic/firmware/discussions/5606

It would be nice if you mention your changes there.

I agree that a double flash solution is not perfect, but on the other side you do not document your suggestion.

Before a brick my t-deck, I might ask for some background :) 

I mention this there too, JIC.

A+
F

**Reply by mverch67** [reply]
This information is outdated and incorrect. Howto's to compile the TFT PR branch can be found on the meshtastic discord channel.

**Reply by mverch67** [reply]
btw: would you be willing to do a Greek translation of the device-ui?

**Reply by iliaschat** [reply]
> This information is outdated and incorrect. Howto's to compile the TFT PR branch can be found on the meshtastic discord channel.

Thank you for pointing that out, and I appreciate your patience. I’m completely new to this project and mistakenly assumed that all the remotes were updated to the latest versions simultaneously. My intention was only to address a few issues I encountered while trying to compile and test the app, and I sincerely hope my previous post didn’t cause any frustration.

I’ll refer to the updated instructions on the Meshtastic Discord channel moving forward. Thank you for guiding me in the right direction, and I apologize for any confusion my post may have caused.

> btw: would you be willing to do a Greek translation of the device-ui?

Sure! 
Here it is: https://github.com/meshtastic/device-ui/pull/33
